### PR TITLE
Bump platform version to a 0.38 adhoc build

### LINKED ISSUE #6128: change services version from 0.37 -> 0.38

### LINKED ISSUE #6206: update services build to latest adhoc 0.38 platform release
Need to incorporate latest features from platform adhoc build 0.38.0-adhoc.xcca9daa7

### PR DESCRIPTION
**Pull request by kimbor**
**Description**:
Update services code to support latest platform build.

There were two platform changes which caused breakage in services:

- https://github.com/hashgraph/hedera-services/pull/6014 VirtualKey is no longer comparable or generic. Since all of our ReadableState classes rely on VirtualKey, this caused cascading effects. Careful review required.
- https://github.com/hashgraph/hedera-services/pull/6099 'stake' is renamed to 'weight'

Fixes #6206  
Fixes #6128

### CODE REVIEW

**Comment by OlegMazurov** [comment]
There two failing unit tests: 
* `ServicesStateTest` fails with OOM due to #5993 
* `SizeLimitedStorageTest` fails with `ClassCastException` due to #6014

**Review by artemananiev — DISMISSED** [review]
Looks fine from VirtualKey/Comparable perspective

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/info/MonoNodeInfo.java
Comment (netopyr) [inline hedera-node/hedera-app/src/main/java/com/hedera/node/app/info/MonoNodeInfo.java:43]: Is this really right? In my undestanding, the term 'weight' was introduced to make the difference between 'stake' and 'weight' more explicit, i.e. there must be a difference. It does not look right, if a method 'isSelfZeroStake()' calls a method 'isSelfZeroWeight()'.

The other question is, if this is really what we want. So far we used this check to ensure that a node without stake submits transactions. Is it really correct to replace stake with weight in this check?

**Comment by kimbor** [comment]
> There are two failing unit tests:
> 
> * `ServicesStateTest` fails with OOM due to [5965: Parallel bucket reads in HalfDiskHashMap.endWriting() #5993](https://github.com/hashgraph/hedera-services/pull/5993)
> * `SizeLimitedStorageTest` fails with `ClassCastException` due to [5910: VirtualKey doesn't need to be comparable #6014](https://github.com/hashgraph/hedera-services/pull/6014)

Fixed.

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/info/MonoNodeInfo.java
Comment (Neeharika-Sompalli) [inline hedera-node/hedera-app/src/main/java/com/hedera/node/app/info/MonoNodeInfo.java:43]: I would think change `public boolean isSelfZeroStake()` also to `public boolean isSelfZeroWeight()`. Or just leave `delegate.isSelfZeroStake()`

File: hedera-node/hedera-mono-service/src/test/java/com/hedera/node/app/service/mono/txns/submission/BasicSubmissionFlowTest.java
Comment (Neeharika-Sompalli) [inline hedera-node/hedera-mono-service/src/test/java/com/hedera/node/app/service/mono/txns/submission/BasicSubmissionFlowTest.java:104]: Weight is derived from stake. We can set up a node's stake and weight is calculated. So I think `setupNonZeroStakeNode`

Comment (kimbor) [reply hedera-node/hedera-mono-service/src/test/java/com/hedera/node/app/service/mono/txns/submission/BasicSubmissionFlowTest.java]: Fixed

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/info/MonoNodeInfo.java
Comment (kimbor) [reply hedera-node/hedera-app/src/main/java/com/hedera/node/app/info/MonoNodeInfo.java:43]: Fixed

Comment (kimbor) [reply hedera-node/hedera-app/src/main/java/com/hedera/node/app/info/MonoNodeInfo.java:43]: Changed.

File: hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/info/NodeInfo.java
Comment (poulok) [inline hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/info/NodeInfo.java:33]: Was this intended to change to weight? Weight is a platform level concept and should only be seen in the application layer when calculating from stake to provide the platform.

**Review by jeromy-cannon — APPROVED** [review]
from a release engineering perspective I approve the build changes, please wait for other approvers for remaining code before merging.

File: hedera-node/hedera-mono-service/build.gradle.kts
Comment (nathanklick) [inline hedera-node/hedera-mono-service/build.gradle.kts:100]: This should be set here: https://github.com/hashgraph/hedera-services/blob/02298a25e1b5a6bfe9c2a7405e5518d366b24f37/buildSrc/src/main/kotlin/com.hedera.hashgraph.conventions.gradle.kts#L168-L172

```suggestion
```

**Review by nathanklick — CHANGES_REQUESTED** [review]

File: hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/context/NodeInfo.java
Comment (Neeharika-Sompalli) [inline hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/context/NodeInfo.java:75]: All the methods in this call should also specify `isZeroStake` even though we check `address.getWeight() <= 0;`

File: hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/queries/answering/StakeAwareAnswerFlow.java
Comment (Neeharika-Sompalli) [inline hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/queries/answering/StakeAwareAnswerFlow.java:41]: same here `isSelfZeroStake`

File: hedera-node/hedera-mono-service/src/test/java/com/hedera/node/app/service/mono/context/NodeInfoTest.java
Comment (Neeharika-Sompalli) [inline hedera-node/hedera-mono-service/src/test/java/com/hedera/node/app/service/mono/context/NodeInfoTest.java:75]: same here in this class

File: hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/info/NodeInfo.java
Comment (kimbor) [reply hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/info/NodeInfo.java]: Fixed

File: hedera-node/hedera-mono-service/build.gradle.kts
Comment (kimbor) [reply hedera-node/hedera-mono-service/build.gradle.kts]: Updated

File: hedera-node/hedera-app/src/test/java/com/hedera/node/app/workflows/ingest/IngestCheckerTest.java
Comment (nickpoorman) [inline hedera-node/hedera-app/src/test/java/com/hedera/node/app/workflows/ingest/IngestCheckerTest.java:155]: Comments should probably say stake.

File: hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/context/NodeInfo.java
Comment (kimbor) [reply hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/context/NodeInfo.java]: Changed.

File: hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/queries/answering/StakeAwareAnswerFlow.java
Comment (kimbor) [reply hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/queries/answering/StakeAwareAnswerFlow.java]: Changed

File: hedera-node/hedera-mono-service/src/test/java/com/hedera/node/app/service/mono/context/NodeInfoTest.java
Comment (kimbor) [reply hedera-node/hedera-mono-service/src/test/java/com/hedera/node/app/service/mono/context/NodeInfoTest.java]: Done. I updated all variables and methods except for `givenEntryWithWeight` and `givenEntryWithMemoAndWeight`. It made sense to me to leave this as `*Weight` because they are using `address.getWeight()`

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/ingest/IngestChecker.java
Comment (Neeharika-Sompalli) [inline hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/ingest/IngestChecker.java:92]:
```suggestion
            // Zero stake nodes are currently not supported
```

File: hedera-node/hedera-app/src/test/java/com/hedera/node/app/workflows/ingest/IngestCheckerTest.java
Comment (Neeharika-Sompalli) [inline hedera-node/hedera-app/src/test/java/com/hedera/node/app/workflows/ingest/IngestCheckerTest.java:157]:
```suggestion
            // Given a node that IS zero stake
```

File: hedera-node/hedera-mono-service/src/test/java/com/hedera/node/app/service/mono/context/NodeInfoTest.java
Comment (Neeharika-Sompalli) [inline hedera-node/hedera-mono-service/src/test/java/com/hedera/node/app/service/mono/context/NodeInfoTest.java:123]:
```suggestion
        givenEntryWithMemoAndStake(nodeId, "Oops!", 0L);
```

File: hedera-node/hedera-app/src/test/java/com/hedera/node/app/workflows/ingest/IngestCheckerTest.java
Comment (kimbor) [reply hedera-node/hedera-app/src/test/java/com/hedera/node/app/workflows/ingest/IngestCheckerTest.java]: Done

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/ingest/IngestChecker.java
Comment (kimbor) [reply hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/ingest/IngestChecker.java]: Done

File: hedera-node/hedera-app/src/test/java/com/hedera/node/app/workflows/ingest/IngestCheckerTest.java
Comment (kimbor) [reply hedera-node/hedera-app/src/test/java/com/hedera/node/app/workflows/ingest/IngestCheckerTest.java]: Changed

File: hedera-node/hedera-mono-service/src/test/java/com/hedera/node/app/service/mono/context/NodeInfoTest.java
Comment (nickpoorman) [reply hedera-node/hedera-mono-service/src/test/java/com/hedera/node/app/service/mono/context/NodeInfoTest.java:123]: This is on the address book. Should we not leave it?

Comment (kimbor) [reply hedera-node/hedera-mono-service/src/test/java/com/hedera/node/app/service/mono/context/NodeInfoTest.java:123]: I updated all other variables and methods except for these two. It made sense to me to leave them as *Weight because they are calling address.getWeight() to determine what value to use.

Comment (Neeharika-Sompalli) [reply hedera-node/hedera-mono-service/src/test/java/com/hedera/node/app/service/mono/context/NodeInfoTest.java:123]: Yeah makes sense

**Review by Neeharika-Sompalli — APPROVED** [review]
LGTM ! Thanks @kimbor

**Review by iwsimon — APPROVED** [review]

**Review by nathanklick — APPROVED** [review]

**Review by nickpoorman — APPROVED** [review]

**Review by qnswirlds — APPROVED** [review]

**Pull request merged**
